lib/mongo/retryable.rb

```diff
@@ -46,18 +36,22 @@ module Retryable
     # @return [ Result ] The result of the operation.
     #
     # @since 2.1.0
-    def read_with_retry(attempt = 0, &block)
```

Comment:
Instead of using recursion to retry the block it now uses the ruby `retry` method. It also passes the exception to the `retry_reconnect` method so that it can log the exception to make the retry visible. It also logs the apparent slowdown during reconnection.